src/solver/optimisation/AreaHydroLevel.cpp

```diff
@@ -0,0 +1,35 @@
+#include "AreaHydroLevel.h"
+
+void AreaHydroLevel::add(int pays, int pdt)
+{
+    const auto& CorrespondanceVarNativesVarOptim
+      = problemeHebdo->CorrespondanceVarNativesVarOptim[pdt];
+    CORRESPONDANCES_DES_CONTRAINTES& CorrespondanceCntNativesCntOptim
+      = problemeHebdo->CorrespondanceCntNativesCntOptim[pdt];
+    CorrespondanceCntNativesCntOptim.NumeroDeContrainteDesNiveauxPays[pays]
+      = problemeHebdo->ProblemeAResoudre->NombreDeContraintes;
+    if (problemeHebdo->CaracteristiquesHydrauliques[pays].SuiviNiveauHoraire)
+    {
+        builder.updateHourWithinWeek(pdt).HydroLevel(pays, 1.0);
+        if (pdt > 0)
+        {
+            builder.updateHourWithinWeek(pdt - 1).HydroLevel(pays, -1.0);
```

Comment:
For example here, why not use a `shift` instead of changing the "base" hour ?
It seems dangerous. Fortunately we restore the timestep later, but we should not need to do this.

Reply:
right, we should use *shift*

Reply:
please [see ](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1607#discussion_r1333000184)